src/zabapgit_services_trans_branch.prog.abap

```diff
@@ -0,0 +1,125 @@
+*&---------------------------------------------------------------------*
+*&  Include  zabapgit_services_trans_branch
+*&---------------------------------------------------------------------*
+CLASS lcl_transport_to_branch DEFINITION.
+  PUBLIC SECTION.
+    METHODS:
+      create
+        IMPORTING io_repository          TYPE REF TO lcl_repo_online
+                  is_transport_to_branch TYPE ty_transport_to_branch
+                  it_transport_objects   TYPE scts_tadir
+        RAISING   lcx_exception.
+  PRIVATE SECTION.
+
+    METHODS create_new_branch
+      IMPORTING
+        io_repository  TYPE REF TO lcl_repo_online
+        iv_branch_name TYPE string
+      RAISING
+        lcx_exception.
+    METHODS add_new_and_changes_to_stage
+      IMPORTING
+        it_transport_objects TYPE scts_tadir
+        io_stage             TYPE REF TO lcl_stage
+        is_stage_objects     TYPE ty_stage_files
+      RAISING
+        lcx_exception.
+    METHODS generate_commit_message
+      IMPORTING
+        is_transport_to_branch TYPE ty_transport_to_branch
+      RETURNING
+        VALUE(rs_comment)      TYPE ty_comment.
+ENDCLASS.
+
+CLASS lcl_transport_to_branch IMPLEMENTATION.
+
+  METHOD create.
+    DATA:
+      ls_transport_object TYPE LINE OF scts_tadir,
+      lt_items            TYPE ty_files_item_tt,
+      ls_local_file       TYPE LINE OF ty_files_item_tt,
+      ls_remote_file      TYPE LINE OF ty_files_tt,
+      ls_item             TYPE string,
+      lv_branch_name      TYPE string,
+      ls_comment          TYPE ty_comment,
+      lo_stage            TYPE REF TO lcl_stage,
+      ls_stage_objects    TYPE ty_stage_files,
+      ls_branch_to_delete TYPE lcl_git_branch_list=>ty_git_branch.
+
+    lv_branch_name = lcl_git_branch_list=>complete_heads_branch_name(
+        lcl_git_branch_list=>normalize_branch_name( is_transport_to_branch-branch_name ) ).
+
+    create_new_branch(
+      io_repository  = io_repository
+      iv_branch_name = lv_branch_name ).
+
+    CREATE OBJECT lo_stage
+      EXPORTING
+        iv_branch_name = lv_branch_name
+        iv_branch_sha1 = io_repository->get_sha1_remote( ).
+
+    ls_stage_objects = lcl_stage_logic=>get( io_repository ).
+
+    add_new_and_changes_to_stage(
+      it_transport_objects = it_transport_objects
+      io_stage             = lo_stage
+      is_stage_objects     = ls_stage_objects ).
+
+    "Stage remote files as removal, for they could be deleted
+    LOOP AT ls_stage_objects-remote INTO ls_remote_file.
+      lo_stage->rm(
```

Comment:
Ah ! Hmmm, interesting problem. Well, that is certainly not `rm` for all remotes. RM will actually delete files from repo.

I think it is better to use lcl_repo->status() instead of stage_logic. It has all the data in a convenient and unified form. Have a look into `lcl_repo_content_browser->list(), build_repo_items_online(), filter_changes()` - it prepares the data for main page 

I think the logic should be:
- loop through transport list
- if `delflag = false` -> find the file, ensure local status is M or A and  `stage->add`
- if `delflag = true` -> find the file, ensure local status is D and `stage-rm`
- I would also put a lot of assertions around just to catch a bug if there is one ... like we assert to find the file with status XXX, if not - exception.

conseqently all can be done in what is currently called `add_new_and_changes_to_stage`.

P.S. if I understand the `delflag` well. I'm not an expert in internals :)